### PR TITLE
Generating header

### LINKED ISSUE #2: Create per-supervision LaTeX file containing information to correctly populate header fields in LaTeX template

### PR DESCRIPTION
**Pull request by AnanyaHN27**
This has been tested

### CODE REVIEW

File: backend/generating/ah2096_DigitalElectronics_1.tex [hunk 1]
```diff
@@ -0,0 +1,8 @@
+\newcommand{\studentname}{Ananya Hari Narain}
+\newcommand{\studentemail}{ah2096@cam.ac.uk}
+\newcommand{\svrname}{Matthew Ireland}
+\newcommand{\svvenue}{Churchill}
+\newcommand{\svcourse}{Digital Electronics}
+\newcommand{\svnumber}{1}
+\newcommand{\svdate}{19/10/19}
+\newcommand{\svtime}{12:30}
```
Comment (MatthewIreland) [inline backend/generating/ah2096_DigitalElectronics_1.tex:8]: This file doesn't need to be included in the repository.

It seems as if this code is generating the headers correctly, but the LaTeX template hasn't been modified to actually use these headers. I suggest we output a file `per_supervision_headers.tex` and then `\input{per_supervision_headers.tex}` in the LaTeX template (we can include an example per_supervision_headers.tex in the template directory if we like; the current example is below).

```
% Configure these parameters.
% Name and email
\newcommand{\studentname}{Harry Potter}
\newcommand{\studentemail}{hjp1@cam.ac.uk}
\newcommand{\svvenue}{Churchill 1C}
\newcommand{\svrname}{Dr John K.~Fawcett}

% Details of supervision
\newcommand{\svcourse}{CST Part IA: Defence Against the Dark Arts}
\newcommand{\svnumber}{1}
\newcommand{\svdate}{2018-10-07}
\newcommand{\svtime}{14:00}
```

File: backend/generating/SupervisionTemplate.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class SupervisionTemplate{
+
+    private SupervisionInfo supervisionInfo;
+    private SupervisionQuestions questions;
```
Comment (MatthewIreland) [inline backend/generating/SupervisionTemplate.java:11]: What is SupervisionQuestions? I don't think this is included in this pull request?

File: backend/generating/Main.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        String studentEmail = null;
+        String svrName = null;
+        String svVenue = null;
+        String svCourse = null;
+        String svNumber = null;
+        String svDate = null;
+        String svTime = null;
+
+        SupervisionInfo supervisionInfo = new SupervisionInfo(studentName, studentEmail, svrName, svVenue,svCourse, svNumber, svDate, svTime);
+        SupervisionQuestions questionList = new SupervisionQuestions(new LinkedList<Question>());
```
Comment (MatthewIreland) [inline backend/generating/Main.java:21]: Can't see SupervisionQuestions in this pull request.

Comment (MatthewIreland) [reply backend/generating/Main.java:21]: Actually, in a previous attempt, we called `SupervisionQuestions` a `SupervisionAgenda` and we called a `Question` a `Task`. I quite like that naming convention.

File: backend/generating/SupervisionTemplate.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        studentData.put("studentname", supervisionInfo.getStudentName());
+        studentData.put("studentemail", supervisionInfo.getStudentEmail());
+        studentData.put("svrname", supervisionInfo.getSvrName());
+        studentData.put("svvenue", supervisionInfo.getSvVenue());
+        studentData.put("svcourse", supervisionInfo.getSvCourse());
+        studentData.put("svnumber", supervisionInfo.getSvNumber());
+        studentData.put("svdate", supervisionInfo.getSvDate());
+        studentData.put("svtime", supervisionInfo.getSvTime());
+        try {
+            create(studentData);
```
Comment (MatthewIreland) [inline backend/generating/SupervisionTemplate.java:31]: Why can't the create method get this info directly from `supervisionInfo`? Is there any reason we need to copy all this data again?

File: backend/generating/SupervisionTemplate.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    private SupervisionQuestions questions;
+
+
+    public SupervisionTemplate(SupervisionInfo supervisionInfo, SupervisionQuestions questions) {
+        //these are the things that the user has to specify when modifying the template
+        this.supervisionInfo = supervisionInfo;
+        this.questions = questions;
+    }
+
+    public void setSupervisionHeader(){
```
Comment (MatthewIreland) [inline backend/generating/SupervisionTemplate.java:20]: Seems like a confusing public API. Do you require users to call this before they call create for the first time?

File: backend/generating/SupervisionTemplate.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        studentData.put("svnumber", supervisionInfo.getSvNumber());
+        studentData.put("svdate", supervisionInfo.getSvDate());
+        studentData.put("svtime", supervisionInfo.getSvTime());
+        try {
+            create(studentData);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    public void create(LinkedHashMap<String, String> studentData) throws IOException {
```
Comment (MatthewIreland) [inline backend/generating/SupervisionTemplate.java:36]: Not a good name. The side effects of this method aren't at all clear from the name alone.

**Review by MatthewIreland — CHANGES_REQUESTED** [review]
We're making progress :)

File: backend/generating/SupervisionTemplate.java (hunk 5 above)
Comment (AnanyaHN27) [reply backend/generating/SupervisionTemplate.java:20]: I've removed the create() method altogether, so I don't think we have this problem anymore.

**Pull request closed without merging**